### PR TITLE
Documenting the igx-grid data structure availability.

### LINKED ISSUE #882: Add to documentation what kind of data structure are available for igx-grid
Most of the examples on igx-grid only allowed the following data structure for it to work perfectly:

```javascript
[
   { header: value, header2: value1, header3, value2, header4: value3 },
   { header: value, header2: value1, header3, value2, header4: value3 },
   { header: value, header2: value1, header3, value2, header4: value3 },
   { header: value, header2: value1, header3, value2, header4: value3 },
   { header: value, header2: value1, header3, value2, header4: value3 },
...and so on 
] 
```

Question is , is this the only data structure it takes?

what if you have something like this? and dont wanna have **all the nested data in 1 column** 

```javascript
 "timeline": {
        "timeline_values": [{
          "tag": 0,
          "start": "11-19"
        },
        {
          "tag": 1,
          "start": "11-20"
        },
        {
          "tag": 2,
          "start": "11-21"
        },
        {
          "tag": 3,
          "start": "11-22"
        }
        ]
      },
      "employees": [{
        "EmployeeID": "56250f",
        "FirstName": "Downs",
        "aval": [
          {"start": "11-19","end": "2","ava": "30","health": "4"},
          {"start": "11-20","end": "2","ava": "40","health": "4"},
          {"start": "11-21","end": "2","ava": "50","health": "4"},
          {"start": "11-22","end": "2","ava": "60","health": "4"}
        ]
      },
      {
        "EmployeeID": "46250f",
        "FirstName": "Mckenzie",
        "aval": [
          {"start": "11-19","end": "2","ava": "1","health": "4"},
          {"start": "11-20","end": "2","ava": "2","health": "4"},
          {"start": "11-21","end": "2","ava": "3","health": "4"},
          {"start": "11-22","end": "2","ava": "4","health": "4"}
        ]
      }, {
        "EmployeeID": "56250f",
        "FirstName": "Charles",
        "aval": [
          {"start": "11-19","end": "2","ava": "100","health": "4"},
          {"start": "11-20","end": "2","ava": "200","health": "4"},
          {"start": "11-21","end": "2","ava": "300","health": "4"},
          {"start": "11-22","end": "2","ava": "400","health": "4"}
        ]
      }
      ]
    }
```
while I solved the issue [here](https://github.com/IgniteUI/igniteui-angular/issues/3046), we would need to know what type of data structure igx-grid takes (in the documentation) and not just guess or assume it will take what the example shows.

### PR DESCRIPTION
**Pull request by HristoP96**
Closes #882

### CODE REVIEW

**Comment by ddincheva** [comment]
@HristoP96 the change seems good, but can you please check if necessary to add these changes and in the Japanese and Korean versions .

**Comment by HristoP96** [comment]
> @HristoP96 the change seems good, but can you please check if necessary to add these changes and in the Japanese and Korean versions .

Done

File: en/components/grid.md [hunk 1]
```diff
@@ -190,6 +190,45 @@ public initColumns(column: IgxGridColumn) {
...
+        ObjectKey2: value2,
+        .
+        .
+        .
+        ObjectKeyN: valueN 
+    }];
+
+```
+>[!WARNING]
+>**The key values must not contain neither arrays or other objects!**.
```
Comment (zdrawku) [inline en/components/grid.md]:
```suggestion
>**The key values must not contain neither arrays or other objects**.
```
Could you please remove the "!" signs

File: en/components/grid.md [hunk 2]
```diff
@@ -190,6 +190,45 @@ public initColumns(column: IgxGridColumn) {
...
+
+>If you use [autoGenerate]({environment:angularApiUrl}/classes/igxgridcomponent.html#autogenerate) columns **the data keys must be identical!**
```
Comment (zdrawku) [inline en/components/grid.md]:
```suggestion
>If you use [autoGenerate]({environment:angularApiUrl}/classes/igxgridcomponent.html#autogenerate) columns **the data keys must be identical.** 
```

File: jp/components/grid.md [hunk 3]
```diff
@@ -190,6 +190,45 @@ public initColumns(column: IgxGridColumn) {
...
+        .
+        .
+        ObjectKeyN: valueN 
+    }];
+
+```
+>[!WARNING]
+>**The key values must not contain neither arrays or other objects!**.
+
+>If you use [autoGenerate]({environment:angularApiUrl}/classes/igxgridcomponent.html#autogenerate) columns **the data keys must be identical!**
```
Comment (zdrawku) [inline jp/components/grid.md]:
```suggestion
>If you use [autoGenerate]({environment:angularApiUrl}/classes/igxgridcomponent.html#autogenerate) columns **the data keys must be identical.**
```

File: jp/components/grid.md [hunk 4]
```diff
@@ -190,6 +190,45 @@ public initColumns(column: IgxGridColumn) {
...
+        ObjectKey2: value2,
+        .
+        .
+        .
+        ObjectKeyN: valueN 
+    }];
+
+```
+>[!WARNING]
+>**The key values must not contain neither arrays or other objects!**.
```
Comment (zdrawku) [inline jp/components/grid.md]:
```suggestion
>**The key values must not contain neither arrays or other objects**.
```

File: kr/components/grid.md [hunk 5]
```diff
@@ -194,6 +194,45 @@ public initColumns(column: IgxGridColumn) {
...
+        ObjectKey2: value2,
+        .
+        .
+        .
+        ObjectKeyN: valueN 
+    }];
+
+```
+>[!WARNING]
+>**The key values must not contain neither arrays or other objects!**.
```
Comment (zdrawku) [inline kr/components/grid.md]:
```suggestion
>**The key values must not contain neither arrays or other objects**.
```

File: kr/components/grid.md [hunk 6]
```diff
@@ -194,6 +194,45 @@ public initColumns(column: IgxGridColumn) {
...
+
+>If you use [autoGenerate]({environment:angularApiUrl}/classes/igxgridcomponent.html#autogenerate) columns **the data keys must be identical!**
```
Comment (zdrawku) [inline kr/components/grid.md]:
```suggestion
>If you use [autoGenerate]({environment:angularApiUrl}/classes/igxgridcomponent.html#autogenerate) columns **the data keys must be identical**
```

**Pull request merged**